### ISSUE TITLE
Scroll/select symbol in real-time

### ISSUE THREAD

**Issue by oin** [darlal/obsidian-switcher-plus#24]
To help fast keyboard-based navigation, well-known symbol switchers like the one of Sublime Text and others scroll the editor view to the selected heading in real-time, and (equally important) scroll back to the cursor if the user cancels.
This way, it's very easy to get a glimpse of what we're switching towards, if the title isn't enough.
It provides an additional layer of context that can allow us to find our way much faster.

It would be nice if Quick Switcher++ worked the same way.

More specifically, in Sublime, the heading/symbol is selected in real-time in addition to scrolling, and if the user cancels the selection is restored to what it was before invoking the switcher.
But not selecting the heading/symbol and just scrolling the view to it would be fine too.
Whatever the underlying API allows more easily, I'd say.

(By the way, the Sublime switcher works the same way for files too, but given that Obisidan's core switcher doesn't work that way I guess it'd be a job for Obsidian devs to support this function for quick file switching.
Nevertheless, it could be done in symbols first and it would already be a huge help.)

**Comment by darlal** [comment]
This is technically feasible, but in practice may not be useful within the context of Obsidian due to the UX model that Obsidian uses.

- The switcher is displayed as a "fullscreen  modal", meaning it takes over the whole app. By contrast, Sublime/VSCode switchers, just display a small unobtrusive search box at the top of the screen. The Obsidian switcher being fullscreens means that centered on page, and takes up a large percentage of the screen real estate when showing symbols for any meaningfully sized file. So with the switcher itself centered on screen and taking up 1/2-2/3 of the real estate, I'm not sure how much value previewing would add. On mobile devices, this would not work at all.
- The switcher, when displayed "dims" the background of the app so it's harder to read the content "behind" the switcher

There are different tricks you could take with the UX, but they would infringe quite a bit on the model defined by the devs. You could also just limit the number of items displayed in the switcher to a really small number, I'm not sure preview is worth that price. The built-in switcher used to only show 5 (I think) and people complained it was not enough and the limit was raised to 10. Here's a screenshot of the "format your links" note from the help vault illustrate the points above. This is on a 3840x1600 display. Just some food for thought.
<img width="1309" alt="Screen Shot 2021-08-05 at 12 54 59 PM" src="https://user-images.githubusercontent.com/5191776/128390741-e546bc57-d399-4732-b79f-c4ce242a8f67.png">

**Comment by oin** [comment]
You're right, Obsidian's design makes it a bit harder, but I think this doesn't make the proposed enhancement less useful. I certainly agree that the plugin's visual appearance and behavior shouldn't stick out from the rest, but to me the issues you pointed out don't seem very serious.

First, the Obsidian switcher indeed takes more space than that of Sublime or VS Code (also the case with my [FoldingText plugin](https://github.com/oin/foldingtext-goto)), but it's still a bit more context than nothing.
In addition, you can easily encounter the same problem in those editors if you're not working full-screen (e.g., with tiled windows, which I do constantly, even with Obsidian), but this behavior still proves really useful (at least, to me). Here's a window with a bit more height than a 1/4 1080p screen in Sublime:

<img width="865" alt="image" src="https://user-images.githubusercontent.com/433830/128403313-ccd6d150-75a9-4a33-8e31-bcec11e4a1c3.png">

As for the dimming, it makes things a bit less visible, but it's not as if it was unreadable. I'm using the _Californa Coast_ theme, and the dimming isn't even that pronounced.

Finally, those inconveniences can easily be removed by any user who wishes, on their own vault, with a trivial CSS snippet like:

```css
.modal-bg {
	background: none;
}
.prompt {
	max-height: 50vh;
}
```

**Comment by darlal** [comment]
Thanks. yea they're not insurmountable, just the UX issue will need to be thought through.
